### ISSUE TITLE
Possible to rewrite BadRequest response contents?

### ISSUE THREAD

**Issue by agjendem** [python-restx/flask-restx#72]
**Ask a question**
Hi! 

This might very well be by design or my incorrect usage of the framework, but I expected to be able to use an error handler to rewrite the response generated when a request fails input validation as defined by my @api.expect()-model.

I'm replacing a system, and I'm trying to keep the responses as close to the old systems format as possible.

I can see while debugging that my custom error handler is triggered, but it seems as there is a "data"-field on the BadRequest exception, that my response in the end is disregarded and the original message is returned to the client? (https://github.com/python-restx/flask-restx/blob/master/flask_restx/api.py#L700)

**Additional context**
Code to reproduce the issue:
```
from flask import Flask, request
from flask_restx import Resource, Api, fields


app = Flask(__name__)
app.config['PROPAGATE_EXCEPTIONS'] = False
api = Api(app)


# Actual validation error: werkzeug.exceptions.BadRequest
@api.errorhandler(Exception)
def handle_validation_error(error):
    return {'my custom': 'error message'}, 400


resource_fields = api.model('Resource', {
    'name': fields.String,
})


@api.route('/hello')
class HelloResource(Resource):
    @api.expect(resource_fields, validate=True)
    def post(self):
        json_data = request.get_json()
        return f'Hello {json_data["name"]}', 200


if __name__ == '__main__':
    app.run(debug=True)
```

* Invalid request that I hoped should return my custom error message as json:
```
curl --data '{"name": 42}' -H 'Content-Type: application/json' http://127.0.0.1:5000/hello
```

Actual response:
```
{
    "errors": {
        "name": "42 is not of type 'string'"
    },
    "message": "Input payload validation failed"
}
```
Expected response:
```
{'my custom': 'error message'}
```

* Valid request (for reference):
```
curl --data '{"name": "Anders"}' -H 'Content-Type: application/json' http://127.0.0.1:5000/hello
```

**Comment by plowman** [comment]
There is probably a better way to do this, but I had a similar struggle in my app and ended up using custom exception classes. Something like this:

```python
import logging

from flask import Flask, request
from flask_restx import Resource, Api, fields
from werkzeug.exceptions import BadRequest

app = Flask(__name__)
api = Api(app)

resource_fields = api.model('Resource', {
  'name': fields.String,
})


class APIException(Exception):
  status_code = 500
  message = 'Unexpected Error'


class BadRequestError(APIException):
  status_code = 400
  message = 'That Request Was Not Good'


@api.errorhandler(APIException)
def handle_bad_request(error):
  response = {
    'error_class': error.__class__.__name__,
    'message': error.message,
  }

  return response, error.status_code


def _handle_error_validation(model):
  try:
    model.validate(request.get_json())
  except BadRequest as exc:
    logging.error(exc, exc_info=True)
    raise BadRequestError
  except Exception as exc:
    logging.error(exc, exc_info=True)
    raise APIException


@api.route('/hello')
class HelloResource(Resource):
  @api.expect(resource_fields)
  def post(self):
    _handle_error_validation(resource_fields)
    json_data = request.get_json()
    return f'Hello {json_data["name"]}'


if __name__ == '__main__':
  app.run(debug=True)
```

I believe this passes your error and valid cases, but let me know if not.

**Comment by agjendem** [comment]
Thanks @plowman, that would work! It's quite close to how we do validation today. I'll probably go with your solution for now!

After I posted this yesterday, I also realized that I would be able to do another dirty workaround:
```
@api.errorhandler(BadRequest)
def handle_validation_error(error):
    api.logger.error(error.data)
    error.data = {'my custom': 'error message'}
    return {}, 400
```

I'll close this for now as there are workarounds. If there are other ways of achieving this, or if this is something that should've worked, I would be interested in contributing to a smoother permanent solution.
